src/blog/2022/07/flowforge-07-released.md

```diff
@@ -0,0 +1,70 @@
+---
+title: FlowForge 0.7 released
+subtitle: Rollbacks, Device Environment Variables and a FlowForge Theme
+description: Rollbacks, Device Environment Variables and a FlowForge Theme
+date: 2022-07-07 12:00:00.0
+authors: ["sam-machin"]
+video: BBNzFInc_2E
+---
+Rollback projects to a previous snapshot, Improvements in using Devices and more.
+
+<!--more-->
+[TODO] Update video url and time.
+
+Keep reading for  the details of whats in this release our you can watch our 1 min roundup video of the new release above. 
+
+We're pleased to announce version 0.7 is now available. the next release of the FlowForge application.
+
+## Features
+[Rollback](https://github.com/flowforge/flowforge/issues/587)
+FlowForge is about running Node-RED at any scale, part of that scale is having multiple users collaborate on the same project. When you are collaborating with people it's important to be able to go back in time to a known working state. As part of that we are introducing rollbacks, this means that you can now take a snapshot of your project at a point in time and then make changes safe in the knowledge that you can rollback to that previous snapshot if you need to.
+
+[Device Environment Variables](https://github.com/flowforge/flowforge/issues/680)
+In the last release we introduced the concept of devices, we're already learning from how these are used and one feature we've added in 0.7 is Device Environment Variables, you have been able to set Environment Variables at the project level for a while now but when deploying a snapshot to multiple devices you may want to override these values for each device, eg to set a site ID or something, this feature enables that.
+
+[FlowForge Theme](https://github.com/flowforge/flowforge-nr-theme/)
+Now that we have a stronger visual identity in the Forge application we have continued that work through to the Node-RED editor. If you are running a Node-RED 3.0 stack you will see a different theme in the editor. It's still very much Node-RED but just has some subtle hints to tie it back to the FlowForge application. We will continue to iterate on this to further integrate the experience between FlowForge and Node-RED in both directions.
+![](../images/ff-07-theme.png)
+
+[ProjectTypes](https://github.com/flowforge/flowforge/issues/380)
+The introduction of ProjectTypes is largely an internal feature to support our FlowForge Cloud proposition by being able to have different price points for projects based on the capacity and features available. While Stacks were a part of this ProjectTypes allows for an aggregation of stacks so that users can change versions of Node-RED while remaining on the same pricing tier.
```

Comment:
If it's just a pricing differentiator feature that should be removed IMO, and when we 'release' the new tier we blog about that. For users, either administrators or end-users this paragraph doesn't influence their life, right?

Reply:
I think it is worth mentioning, but the current wording (with so much emphasis on it being a 'largely internal feature') does perhaps miss the value to end-users (admins).

> The introduction of ProjectTypes is a way to group Stacks together that share common characteristics - such as memory/cpu limits, or the availability of particular features. In platforms with billing enabled, such as our own FlowForge Cloud, the ProjectTypes can have different price points set on them. Within FlowForge Cloud, you'll see we've introduced the Small ProjectType - which applies to all existing projects on the platform.

Reply:
agree Nick focussing on the feature first and then how we're utilising it in ff cloud is a good approach to this kind of thing.

Reply:
Makes sense @knolleary . Thanks